gson/src/main/java/com/google/gson/NewlineStyle.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2022 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gson;
+
+/**
+ * An enumeration that defines the kind of newline to use for serialization.
+ *
+ * @see <a href="https://en.wikipedia.org/wiki/Newline">Wikipedia Newline article</a>
+ *
+ * @since $next-version$
+ */
+public enum NewlineStyle {
+  /**
+   * Using this style will result in the same kind of newline that the current environment uses.
+   *
+   * <p>So it will produce {@code "\r\n"} when running on Windows, and {@code "\n"} when running
+   * on macOS &amp; Linux.</p>
+   *
+   * <p>It might be good discipline to not use {@code CURRENT_OS} when exchanging JSON data
+   * between systems, or storing them long term.<br>
+   * Gson will parse any valid JSON regardless the newline style.<br>
```

Comment:
Really minor, but maybe the `<br>` here could be omitted since the next sentence is still closely related to the previous one.
```suggestion
   * Gson will parse any valid JSON regardless the newline style.
```